HTML nesting error: Button nested inside button in StructuresTab

## Bug Description

In the Protocol Editor's Structures tab, there's an HTML nesting error where a `<button>` element is nested inside another `<button>` element. This violates HTML semantics and causes React hydration errors.

## Console Errors

```
[ERROR] In HTML, %s cannot be a descendant of <%s>.
This will cause a hydration error.%s <button> button

[ERROR] <%s> cannot contain a nested %s. button <button>
```

## Reproduction Steps

1. Navigate to Protocol Editor (e.g., `/protocols/proto-at-commands/edit`)
2. Click on the "Structures" tab
3. Create a new structure by clicking "Add Structure"
4. Observe the console errors about button nesting

## Stack Trace (from console)

```
StructuresTab editState={{id:"proto-...", ...}} onChange={function handleChange} ...>
  <div className="space-y-6">
    <div>
    <div className="space-y-2">
      <div className="border bor...">
        <button onClick={function onClick} className="w-full flex items-center gap-3 p-4 text-left hover:bg-muted/50">
          <ChevronRight>
          <div>
          <_c variant="ghost" size="icon" className="h-8 w-8" onClick={function onClick} ...>
            <button className="inline-flex items-center..." onClick={function onClick} title="Edit Structure">
```

## Root Cause

The structure row is using a `<button>` as the clickable container, and the "Edit Structure" action button inside it is also a `<button>`. This creates invalid HTML nesting.

## Expected Behavior

The outer clickable element should use a `<div>` with `role="button"` and proper keyboard handling, or the inner edit button should be positioned absolutely outside the button flow.

## Suggested Fix

Option 1: Change the outer container from `<button>` to `<div role="button" tabIndex={0}>` with proper keyboard event handling.

Option 2: Use event.stopPropagation() on inner buttons and restructure the component to avoid nesting.

## Environment

- Browser: Chrome (running in Vite dev mode)
- URL: `http://localhost:14201/protocols/proto-at-commands/edit` (Structures tab)
- Component: `src/pages/ProtocolEditor/tabs/StructuresTab.tsx`

## Screenshot

The issue occurs when structures are created and displayed in the list.

---
🤖 Generated with [Claude Code](https://claude.ai/code)

--- a/src/pages/ProtocolEditor/tabs/StructuresTab.tsx
+++ b/src/pages/ProtocolEditor/tabs/StructuresTab.tsx
@@ -136,11 +136,21 @@ export const StructuresTab: React.FC<StructuresTabProps> = ({
               key={structure.id}
               className="border border-border rounded-lg overflow-hidden"
             >
-              <button
+              <div
+                role="button"
+                tabIndex={0}
                 onClick={() =>
                   setExpanded(expanded === structure.id ? null : structure.id)
                 }
-                className="w-full flex items-center gap-3 p-4 text-left hover:bg-muted/50"
+                onKeyDown={(e) => {
+                  if (e.key === "Enter" || e.key === " ") {
+                    e.preventDefault();
+                    setExpanded(
+                      expanded === structure.id ? null : structure.id,
+                    );
+                  }
+                }}
+                className="w-full flex items-center gap-3 p-4 text-left hover:bg-muted/50 cursor-pointer"
               >
                 {expanded === structure.id ? (
                   <ChevronDown className="w-4 h-4" />
@@ -176,7 +186,7 @@ export const StructuresTab: React.FC<StructuresTabProps> = ({
                 >
                   <Trash2 className="w-4 h-4 text-destructive" />
                 </Button>
-              </button>
+              </div>
               {expanded === structure.id && (
                 <div className="border-t border-border p-4 bg-muted/30">
                   <div className="flex items-center justify-between mb-3">

